lib/src/utils/commands_extension.dart

```diff
@@ -21,31 +21,38 @@ import 'dart:convert';
 
 import 'package:matrix/matrix.dart';
 
+/// callback taking [CommandArgs] as input and a [StringBuffer] as standard output
+/// optionally returns an event ID as in the [Room.sendEvent] syntax.
+/// a [CommandException] should be thrown if the specified arguments are considered invalid
+typedef CommandExecutionCallback = FutureOr<String?> Function(
+  CommandArgs,
+  StringBuffer? stdout,
+);
+
 extension CommandsClientExtension on Client {
   /// Add a command to the command handler. `command` is its name, and `callback` is the
   /// callback to invoke
-  void addCommand(
-    String command,
-    FutureOr<String?> Function(CommandArgs) callback,
-  ) {
+  void addCommand(String command, CommandExecutionCallback callback) {
     commands[command.toLowerCase()] = callback;
   }
 
   /// Parse and execute a string, `msg` is the input. Optionally `inReplyTo` is the event being
   /// replied to and `editEventId` is the eventId of the event being replied to
   Future<String?> parseAndRunCommand(
```

Comment:
Can we also add a dartdoc comment here saying "room object can be null here only when you are very sure the msg doesn't contain room specific commands"